### PR TITLE
[Exam] Quiz component for exams

### PR DESCRIPTION
**Pull request by filip-gregurevic**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I documented the TypeScript code using JSDoc style.

### Motivation and Context

The current implementation of the quiz exercises for the different modes includes too many features which are not needed / shouldn't be available during an exam.

### Description

Extracted the base functionality of a quiz participation into a new component.
The participation is now an input of the component instead of being loaded via the url param.
The bar with the quiz-step icons is displayed on the left side and it can be navigated to the step by clicking it.

### Steps for Testing

**can only be tested locally until student exam participations aren't loaded into the exam view**

Setup:
1. Import and embed the component into a component which contains a quiz participation (e.g. the quiz component). You can also comment out the other component so that only the one to test is displayed.
2. Pass the participation into the quiz-exam-component.

Test:
1. Log in to Artemis
2. Create a quiz with every type of quiz question
3. Navigate to where the component is displayed (e.g. a quiz open for practice)
4. Check that the quiz' components are displayed correctly and that you can interact with them
5. Check that you can navigate to a specific quiz question via the bottom bar

### Screenshots

Bildschirmfoto 2020-06-18 um 15 48 47
Bildschirmfoto 2020-06-18 um 15 48 36
Bildschirmfoto 2020-06-18 um 15 48 28

### CODE REVIEW

File: src/main/webapp/app/exam/participate/exercises/quiz/quiz-exam-participation.component.ts [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+            shortAnswerSubmittedAnswer.quizQuestion = question;
+            shortAnswerSubmittedAnswer.submittedTexts = this.shortAnswerSubmittedTexts[questionID];
+            this.submission.submittedAnswers.push(shortAnswerSubmittedAnswer);
+        }, this);
+    }
+
+    /**
+     * Apply the data of the participation, replacing all old data
+     */
+    updateParticipationFromServer(participation: StudentParticipation) {
```
Comment (krusche) [inline src/main/webapp/app/exam/participate/exercises/quiz/quiz-exam-participation.component.ts:224]: I think we should rename this method as the participation is probably not coming from the server directly

Comment (filip-gregurevic) [reply src/main/webapp/app/exam/participate/exercises/quiz/quiz-exam-participation.component.ts]: I renamed the method now to only _updateParticipation_

**Comment by fde312** [comment]
I tested it as well locally and everything is working fine. Good job 👍 

image

Just one small issue. For me the left quiz-bar is over the quiz-exam-footer (both have z-index: 10). We should change the one of the quiz-bar

File: src/main/webapp/app/exam/participate/exercises/quiz/quiz-exam-participation.component.scss [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+.quiz-footer {
+    width: 60px;
+    height: 100%;
+    position: absolute;
+    padding: 0 10px;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    z-index: 10;
```
Comment (fde312) [inline src/main/webapp/app/exam/participate/exercises/quiz/quiz-exam-participation.component.scss:20]:
```suggestion
    z-index: 8;
```

**Pull request merged**